spi/src/main/java/org/jboss/pnc/spi/coordinator/BuildSetTask.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.spi.coordinator;
+
+import org.jboss.pnc.model.BuildConfigSetRecord;
+import org.jboss.pnc.model.BuildConfiguration;
+import org.jboss.pnc.model.BuildConfigurationSet;
+import org.jboss.pnc.model.ProductMilestone;
+import org.jboss.pnc.spi.BuildSetStatus;
+import org.jboss.pnc.spi.events.BuildSetStatusChangedEvent;
+
+import java.util.Date;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * @author Tomas Remes
+ */
+public interface BuildSetTask {
```

Comment:
I guess we would need to move `DefaultBuildStatusChangedEvent` and then it could be possible. Should I give it a try?